pkg/apis/tenancy/v1alpha1/types.go

```diff
@@ -210,6 +210,10 @@ type WorkspaceShardStatus struct {
 	// Connection information for the WorkspaceShard.
 	// +optional
 	ConnectionInfo *ConnectionInfo `json:"connectionInfo,omitempty"`
+
+	// Version of credentials last successfully loaded.
+	// +optional
+	CredentialVersion string `json:"credentialVersion,omitempty"`
```

Comment:
I wanted something less sensitive than the resource version of the secret - really detect when the underlying kubeconfig changed.

Reply:
A hash makes sense 👍

Reply:
Might want to call it CredentialsHash instead of Version

Reply:
I kind of wanted it to be opaque, not sure it's important to know that the credentials were hashed to create it. WDYT?